Completed the README and Code Documentation

beginner003 opened the pull request:
Second time in merging, will request review this time.

beginner003 committed:
Update README.md

Edited the README

beginner003 committed:
Create CodeDocumentation.md

The code documentation is created to explain the code structure and methods

beginner003 committed:
Uploaded Cover Image For README

This image is for the README cover. It is generated by Adobe Firefly with the prompt "The Launch Interceptor Project implements a function called DECIDE() that determines whether an interceptor should be launched based on input radar tracking information. for anti-ballistic missile systems, providing a systematic approach to decision-making based on real-time radar data. 4K, sci-fi"

beginner003 committed:
Merge branch 'main' into Yoyo/README/add1

edgarwolff commented on Cover.jpg:
Maybe put it in a sudfolder for a cleaner repo :man_shrugging:

edgarwolff commented on README.md:
```diff
@@ -1 +1,78 @@
+        - Designed and Implemented the LIC06, LIC07, LIC08 methods.
+        - Designed and Implemented the Helper Function `pointLineDistance()`
+        - Created unit tests for LIC06, LIC07, LIC08 methods, and the above helper function.
+- [**C](https://github.com/anotherusername)heang Weng Io, Yoyo (**[https://github.com/beginner003](https://github.com/beginner003))
```
What is the link to https://github.com/anotherusername made for?

beginner003 replied:
Fixed it! Thanks

edgarwolff commented on CodeDocumentation.md:
```diff
@@ -0,0 +1,553 @@
+The project consists of two primary Java files:
+
+1. **LaunchInterceptor.java**: This file contains the implementation of the `LaunchInterceptor` class.
+2. **LaunchInterceptorTEST.java**: This file includes unit tests for the methods defined in the `LaunchInterceptor` class.
```
See commit [https://github.com/blimpan/DD2480-SWEF/pull/93/commits/01f119cd0d9c3999a03a3b55f324ce0ad02f5845](01f119c)
Filename got changed to LaunchInterceptorTest to match with java naming conventions

beginner003 replied:
Fixed! Thanks!

edgarwolff commented on CodeDocumentation.md:
```diff
@@ -0,0 +1,553 @@
+- `processed`: Indicates if the decide method has been processed.
+- `PUM`: Preliminary Unlocking Matrix.
+- `CMV`: Conditions Met Vector.
+- `FUV`: Final Unlocking Vector.
```
FUV gets processed by the method rather then used, it already existed and is just filled by the method.
Technically it would be more the LCM and the PUV which are used and the 3 above gets defined.
Also the processed boolean is more for an "under the hood" purpose and would maybe not be worth mentionning

beginner003 replied:
I amended the description based on your review, please check if it is okay

edgarwolff commented on CodeDocumentation.md:
```diff
@@ -0,0 +1,553 @@
+
+---
+
+### LaunchInterceptorTEST.java
```
Same as above, needs refactoring

edgarwolff commented on CodeDocumentation.md:
```diff
@@ -0,0 +1,553 @@
+
+This file contains unit tests for the `LaunchInterceptor` class. It verifies the functionality of the implemented methods, ensuring accurate evaluations of the LIC conditions and launch decisions based on input parameters and data points. The tests cover:
+
+- **Constructor Testing**:
```
See the branch with the decide method coded I changed these tests to split them and give them a more suitable name

beginner003 replied:
I added one point for the getter testing, please check if it is ok

edgarwolff replied:
The formulation sounds a bit weird "Validate the getter methods..." seems better to me or not?

edgarwolff commented on CodeDocumentation.md:
```diff
@@ -0,0 +1,553 @@
+    - General cases where the condition is satisfied.
+    - Boundary cases with the minimum number of points.
+    - Invalid parameters (e.g., negative `LENGTH1`).
+- **Helper Fucntion Testing**
```
Typo in word "Fucntion"

beginner003 replied:
Fixed! thanks!

edgarwolff commented on CodeDocumentation.md:
```diff
@@ -0,0 +1,553 @@
+    - General cases
+    - Complex computations Cases
+- **PUM (Preliminary Unlocking Matrix)**:
+    - Validating that the PUM is correctly populated based on LIC evaluations.
```
Do such tests exist? (Same for the 2 next)

beginner003 replied:
Resolved! I deleted them and amended the decide method testing documentaiton

edgarwolff reviewed (changes requested):
See comments below for requested changes
Thanks for the documentation which is far more detailled than I would have expected.

WatermelonGodz committed:
Update README.md

Corrected which LICs Henrik and Linus worked on

beginner003 committed:
Created a new folder to store the cover image

beginner003 committed:
Added Cover Image to the file

Stored the image in a file for better readability

beginner003 committed:
Delete Cover.jpg

beginner003 committed:
Fixed typos

Fixed typos
Added link to Code Documentation
Fixed the link the the cover image

beginner003 committed:
Update CodeDocumentation.md based on reviews

- fixed typos
- updated the file naming according to Edgar's branch
- fixed content of the testing code documentation

edgarwolff reviewed (approved):
One or 2 formulations could be enhanced but this is already fine for me

beginner003 committed:
Update README.md

fixed the title md

KristinRosen commented on README.md:
```diff
@@ -1 +1,77 @@
+        - Designed the skeleton of the [`LaunchInterceptorTEST.java`](http://LaunchInterceptorTEST.java)
+        - Designed and Implemented the LIC03, LIC04, LIC05 methods.
+        - Created unit tests for LIC03, LIC04, LIC05 methods.
+- **Kristen Rosen (**[https://github.com/KristinRosen](https://github.com/KristinRosen))
```
*Kristin, not Kristen :)

beginner003 replied:
Fixed! Sorry for the typo~

KristinRosen commented on README.md:
```diff
@@ -1 +1,77 @@
+- **Kristen Rosen (**[https://github.com/KristinRosen](https://github.com/KristinRosen))
+    - **Contributions**:
+        - Designed and Implemented the LIC06, LIC07, LIC08 methods.
+        - Designed and Implemented the Helper Function `pointLineDistance()`
```
I also did the function "pointsDistance()"

beginner003 replied:
Added! Thanks

KristinRosen reviewed (changes requested):
marked two comments - spelling of my name + the second helper function I made

KristinRosen reviewed (changes requested):
missing description of lic8 which is:

 "Determines if there exists at least one set of three data points separated by exactly A_PTS and B_PTS consecutive intervening points, respectively, that cannot be contained within or on a circle of radius RADIUS1.

Returns true if condition is metor false otherwise"

blimpan reviewed (approved):
As long as Edgar's comments get resolved I am happy with this. Good job!

edgarwolff commented on README.md:
```diff
@@ -1 +1,77 @@
+        - Created unit tests for LIC09, LIC10, LIC11 methods, and the above helper functions.
+- **Edgar Wolff (**[https://github.com/edgarwolff](https://github.com/edgarwolff))
+    - **Contributions**:
+        - Refactored the code infrastructure for easier implementations of LIC methods.
```
Add also the constructor test and the main program file

beginner003 replied:
Added! Thank you!

edgarwolff reviewed (changes requested):
See comments for requested changes

WatermelonGodz commented on CodeDocumentation.md:
```diff
@@ -0,0 +1,547 @@
+
+### Specifications
+
+- Determines whether three points can be contained within a circle of a specified radius. Depending on the mode, it checks for containment or exclusion.
```
Depending on the mode, it checks for if all three points can be contained in a circle or if there is no circle that contains all three points.

beginner003 replied:
Amended with your description! Thanks!

WatermelonGodz reviewed (commented):
A small clarification about containedInCircle, other than that it looks good

beginner003 committed:
Update README.md based on reviews

-fixed typos
-amended the statement of contributions

beginner003 committed:
Update CodeDocumentation.md based on reviews

- Added documentation for LIC8
- Amended the documentation for helper function containInCircle().
